### PR TITLE
Refactor - Using utils

### PR DESCRIPTION
**Pull request by sbeih**
- `src/utils`: utils for arrays, pq and zip
- `constants`: file pathes
- `generators`: string templates

### CODE REVIEW

**Review by ron-b — CHANGES_REQUESTED** [review]

File: src/utils/zipUtils.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    await JSZipModule.loadAsync(
+        WorkbookTemplate.SIMPLE_QUERY_WORKBOOK_TEMPLATE,
+        {
+            base64: true,
+        }
+    );
+
+const loadAsyncTemplate = async (template: string): Promise<JSZip> =>
+    await JSZipModule.loadAsync(template, {
+        base64: true,
```
[the remark this answers is not in the document]
Comment (sbeih) [reply src/utils/zipUtils.ts:24]: It was passed before my change

[the remark this answers is not in the document]
Comment (ron-b) [reply src/utils/zipUtils.ts:24]: image

File: src/utils/pqUtils.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+const getBase64 = async (zip: JSZip): Promise<string> => {
+    const xmlValue = await zip.file(pqCustomXmlPath)?.async("uint8array");
+    if (xmlValue === undefined) {
+        throw new Error("PQ document wasn't found in zip");
+    }
+    const xmlString = iconv.decode(xmlValue.buffer as Buffer, "UTF-16");
+    const parser: DOMParser = new DOMParser();
+    const doc: Document = parser.parseFromString(xmlString, "text/xml");
+    const result = doc.childNodes[0].textContent;
```
Comment (ron-b) [inline src/utils/pqUtils.ts:17]: nit - maybe fix potential bug here where doc doesn't have childNodes we'll crash with a non descriptive message.

File: src/utils/pqUtils.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import JSZip from "jszip";
```
Comment (ron-b) [inline src/utils/pqUtils.ts:4]: this file is best named pqCustomXmlUtils

**Review by ron-b — APPROVED** [review]

**Pull request merged**